server/controllers/alertsController.ts

```diff
@@ -1,62 +1,174 @@
 import { mapHeaderData } from '../mappers/headerMappers'
-import { PrisonerSearchService } from '../services'
 import { Role } from '../data/enums/role'
-import config from '../config'
-import { userCanEdit, userHasRoles } from '../utils/utils'
-import { Prisoner } from '../interfaces/prisoner'
-import { InmateDetail } from '../interfaces/prisonApi/inmateDetail'
+import { formatName, userCanEdit, userHasRoles } from '../utils/utils'
+import { NameFormatStyle } from '../data/enums/nameFormatStyle'
+import { formatDate } from '../utils/dateHelpers'
+import { AlertForm, AlertType } from '../interfaces/prisonApi/alert'
+import ReferenceDataService from '../services/referenceDataService'
+import { FlashMessageType } from '../data/enums/flashMessageType'
+import { Alert } from '../interfaces/prisoner'
 
 /**
  * Parse request for alerts page and orchestrate response
  */
 export default class AlertsController {
-  private isActive: boolean
-
   constructor(
-    isActive: boolean,
-    private readonly prisonerSearchService: PrisonerSearchService,
     private readonly alertsPageService: AlertsPageService,
-  ) {
-    this.isActive = isActive
+    private readonly referenceDataService: ReferenceDataService,
+  ) {}
+
+  public displayAlerts(isActive: boolean): RequestHandler {
+    return async (req: Request, res: Response, next: NextFunction) => {
+      // Get data from middleware
+      const { prisonerData, inmateDetail } = req.middleware
+
+      // Parse query params for paging, sorting and filtering data
+      const { clientToken } = res.locals
+      const queryParams: PagedListQueryParams = {}
+      if (req.query.page) queryParams.page = +req.query.page
+      if (req.query.sort) queryParams.sort = req.query.sort as string
+      if (req.query.alertType) queryParams.alertType = req.query.alertType as string[]
+      if (req.query.from) queryParams.from = req.query.from as string
+      if (req.query.to) queryParams.to = req.query.to as string
+      if (isActive) queryParams.alertStatus = 'ACTIVE'
+      if (!isActive) queryParams.alertStatus = 'INACTIVE'
+      if (req.query.showAll) queryParams.showAll = Boolean(req.query.showAll)
+
+      // Set role based permissions
+      const canUpdateAlert =
+        userHasRoles([Role.UpdateAlert], res.locals.user.userRoles) && userCanEdit(res.locals.user, prisonerData)
+
+      let addAlertLinkUrl: string
+      if (canUpdateAlert) {
+        addAlertLinkUrl = `/prisoner/${prisonerData.prisonerNumber}/add-alert`
+      }
+
+      // Get alerts based on given query params
+      const alertsPageData = await this.alertsPageService.get(clientToken, prisonerData, queryParams, canUpdateAlert)
+      const showingAll = queryParams.showAll
+
+      // Render page
+      return res.render('pages/alerts/alertsPage', {
+        pageTitle: 'Alerts',
+        ...mapHeaderData(prisonerData, inmateDetail, res.locals.user, 'alerts'),
+        ...alertsPageData,
+        showingAll,
+        addAlertLinkUrl,
```

Comment:
@davidmiddletonmoj, what happens if this is undefined? Does the template handle that?